java/performance/src/test/java/org/apache/arrow/adapter/AvroAdapterBenchmarks.java

```diff
@@ -0,0 +1,140 @@
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+/**
+ * Benchmarks for avro adapter.
+ */
+@State(Scope.Benchmark)
+public class AvroAdapterBenchmarks {
+
+  @ClassRule
+  public static final TemporaryFolder TMP = new TemporaryFolder();
+
+  private final int valueCount = 3000;
+
+  private AvroToArrowConfig config;
+
+  private Schema schema;
+  private Decoder decoder;
+
+  /**
+   * Setup benchmarks.
+   */
+  @Setup
+  public void prepare() throws Exception {
+    BufferAllocator allocator = new RootAllocator(Integer.MAX_VALUE);
+    config = new AvroToArrowConfig(allocator);
+
+    String schemaStr = "{\n" + " \"namespace\": \"org.apache.arrow.avro\",\n" +
+         " \"type\": \"record\",\n" + " \"name\": \"testBenchmark\",\n" + " \"fields\": [\n" +
+         "    {\"name\": \"f0\", \"type\": \"string\"},\n" +
+         "    {\"name\": \"f1\", \"type\": \"int\"},\n" +
+         "    {\"name\": \"f2\", \"type\": \"long\"},\n" +
+         "    {\"name\": \"f3\", \"type\": \"boolean\"},\n" +
+         "    {\"name\": \"f4\", \"type\": \"float\"}\n" + "  ]\n" + "}";
+    schema = new Schema.Parser().parse(schemaStr);
+
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    BinaryEncoder encoder = new EncoderFactory().directBinaryEncoder(out, null);
+    DatumWriter writer = new GenericDatumWriter(schema);
+
+    decoder = new DecoderFactory().directBinaryDecoder(new ByteArrayInputStream(out.toByteArray()), null);
+
+    for (int i = 0; i < valueCount; i++) {
+      GenericRecord record = new GenericData.Record(schema);
+      record.put(0, "test" + i);
+      record.put(1, i);
+      record.put(2, i + 1L);
+      record.put(3, i % 2 == 0);
+      record.put(4, i + 0.1f);
+      writer.write(record, encoder);
+    }
+
+  }
+
+  /**
+   * Tear down benchmarks.
+   */
+  @TearDown
+  public void tearDown() {
+    config.getAllocator().close();
+  }
+
+  /**
+   * Test {@link AvroToArrow#avroToArrowIterator(Schema, Decoder, AvroToArrowConfig)}
+   * @return useless. To avoid DCE by JIT.
+   */
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public int testAvroToArrow() throws Exception {
+    try (AvroToArrowVectorIterator iter = AvroToArrow.avroToArrowIterator(schema, decoder, config)) {
+      while (iter.hasNext()) {
+        VectorSchemaRoot root = iter.next();
```

Comment:
I expect having the side effects on decoder are enough but just in case can you do something with one of the vectors (maybe sum the int) to ensure nothing gets eliminated as dead-code.

Reply:
ok, fixed.